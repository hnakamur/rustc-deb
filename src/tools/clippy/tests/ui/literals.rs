#![warn(clippy::large_digit_groups)]
#![warn(clippy::mixed_case_hex_literals)]
#![warn(clippy::unseparated_literal_suffix)]
#![warn(clippy::zero_prefixed_literal)]
#![allow(dead_code)]

fn main() {
    let ok1 = 0xABCD;
    let ok3 = 0xab_cd;
    let ok4 = 0xab_cd_i32;
    let ok5 = 0xAB_CD_u32;
    let ok5 = 0xAB_CD_isize;
    let fail1 = 0xabCD;
    let fail2 = 0xabCD_u32;
    let fail2 = 0xabCD_isize;
    let fail_multi_zero = 000_123usize;

    let ok6 = 1234_i32;
    let ok7 = 1234_f32;
    let ok8 = 1234_isize;
    let fail3 = 1234i32;
    let fail4 = 1234u32;
    let fail5 = 1234isize;
    let fail6 = 1234usize;
    let fail7 = 1.5f32;

    let ok9 = 0;
    let ok10 = 0_i64;
    let fail8 = 0123;

    let ok11 = 0o123;
    let ok12 = 0b10_1010;

    let ok13 = 0xab_abcd;
    let ok14 = 0xBAFE_BAFE;
    let ok15 = 0xab_cabc_abca_bcab_cabc;
    let ok16 = 0xFE_BAFE_ABAB_ABCD;
    let ok17 = 0x123_4567_8901_usize;

    let fail9 = 0xabcdef;
    let fail10 = 0xBAFEBAFE;
    let fail11 = 0xabcdeff;
    let fail12 = 0xabcabcabcabcabcabc;
    let fail13 = 0x1_23456_78901_usize;

    let fail14 = 2_32;
    let fail15 = 4_64;
    let fail16 = 7_8;
    let fail17 = 23_16;
    let ok18 = 23_128;
    let fail19 = 12_3456_21;
    let fail20 = 2__8;
    let fail21 = 4___16;
    let fail22 = 3__4___23;
    let fail23 = 3__16___23;

    let fail24 = 12.34_64;
    let fail25 = 1E2_32;
    let fail26 = 43E7_64;
    let fail27 = 243E17_32;
    let fail28 = 241251235E723_64;
    let fail29 = 42279.911_32;
}
