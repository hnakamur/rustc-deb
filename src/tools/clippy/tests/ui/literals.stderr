error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:13:17
   |
LL |     let fail1 = 0xabCD;
   |                 ^^^^^^
   |
   = note: `-D clippy::mixed-case-hex-literals` implied by `-D warnings`

error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:14:17
   |
LL |     let fail2 = 0xabCD_u32;
   |                 ^^^^^^^^^^

error: inconsistent casing in hexadecimal literal
  --> $DIR/literals.rs:15:17
   |
LL |     let fail2 = 0xabCD_isize;
   |                 ^^^^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:16:27
   |
LL |     let fail_multi_zero = 000_123usize;
   |                           ^^^^^^^^^^^^
   |
   = note: `-D clippy::unseparated-literal-suffix` implied by `-D warnings`

error: this is a decimal constant
  --> $DIR/literals.rs:16:27
   |
LL |     let fail_multi_zero = 000_123usize;
   |                           ^^^^^^^^^^^^
   |
   = note: `-D clippy::zero-prefixed-literal` implied by `-D warnings`
help: if you mean to use a decimal constant, remove the `0` to remove confusion
   |
LL |     let fail_multi_zero = 123usize;
   |                           ^^^^^^^^
help: if you mean to use an octal constant, use `0o`
   |
LL |     let fail_multi_zero = 0o123usize;
   |                           ^^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:21:17
   |
LL |     let fail3 = 1234i32;
   |                 ^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:22:17
   |
LL |     let fail4 = 1234u32;
   |                 ^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:23:17
   |
LL |     let fail5 = 1234isize;
   |                 ^^^^^^^^^

error: integer type suffix should be separated by an underscore
  --> $DIR/literals.rs:24:17
   |
LL |     let fail6 = 1234usize;
   |                 ^^^^^^^^^

error: float type suffix should be separated by an underscore
  --> $DIR/literals.rs:25:17
   |
LL |     let fail7 = 1.5f32;
   |                 ^^^^^^

error: this is a decimal constant
  --> $DIR/literals.rs:29:17
   |
LL |     let fail8 = 0123;
   |                 ^^^^
help: if you mean to use a decimal constant, remove the `0` to remove confusion
   |
LL |     let fail8 = 123;
   |                 ^^^
help: if you mean to use an octal constant, use `0o`
   |
LL |     let fail8 = 0o123;
   |                 ^^^^^

error: long literal lacking separators
  --> $DIR/literals.rs:40:17
   |
LL |     let fail9 = 0xabcdef;
   |                 ^^^^^^^^ help: consider: `0x00ab_cdef`
   |
   = note: `-D clippy::unreadable-literal` implied by `-D warnings`

error: long literal lacking separators
  --> $DIR/literals.rs:41:18
   |
LL |     let fail10 = 0xBAFEBAFE;
   |                  ^^^^^^^^^^ help: consider: `0xBAFE_BAFE`

error: long literal lacking separators
  --> $DIR/literals.rs:42:18
   |
LL |     let fail11 = 0xabcdeff;
   |                  ^^^^^^^^^ help: consider: `0x0abc_deff`

error: long literal lacking separators
  --> $DIR/literals.rs:43:18
   |
LL |     let fail12 = 0xabcabcabcabcabcabc;
   |                  ^^^^^^^^^^^^^^^^^^^^ help: consider: `0x00ab_cabc_abca_bcab_cabc`

error: digit groups should be smaller
  --> $DIR/literals.rs:44:18
   |
LL |     let fail13 = 0x1_23456_78901_usize;
   |                  ^^^^^^^^^^^^^^^^^^^^^ help: consider: `0x0123_4567_8901_usize`
   |
   = note: `-D clippy::large-digit-groups` implied by `-D warnings`

error: mistyped literal suffix
  --> $DIR/literals.rs:46:18
   |
LL |     let fail14 = 2_32;
   |                  ^^^^ help: did you mean to write: `2_i32`
   |
   = note: #[deny(clippy::mistyped_literal_suffixes)] on by default

error: mistyped literal suffix
  --> $DIR/literals.rs:47:18
   |
LL |     let fail15 = 4_64;
   |                  ^^^^ help: did you mean to write: `4_i64`

error: mistyped literal suffix
  --> $DIR/literals.rs:48:18
   |
LL |     let fail16 = 7_8;
   |                  ^^^ help: did you mean to write: `7_i8`

error: mistyped literal suffix
  --> $DIR/literals.rs:49:18
   |
LL |     let fail17 = 23_16;
   |                  ^^^^^ help: did you mean to write: `23_i16`

error: digits grouped inconsistently by underscores
  --> $DIR/literals.rs:51:18
   |
LL |     let fail19 = 12_3456_21;
   |                  ^^^^^^^^^^ help: consider: `12_345_621`
   |
   = note: `-D clippy::inconsistent-digit-grouping` implied by `-D warnings`

error: mistyped literal suffix
  --> $DIR/literals.rs:52:18
   |
LL |     let fail20 = 2__8;
   |                  ^^^^ help: did you mean to write: `2_i8`

error: mistyped literal suffix
  --> $DIR/literals.rs:53:18
   |
LL |     let fail21 = 4___16;
   |                  ^^^^^^ help: did you mean to write: `4_i16`

error: digits grouped inconsistently by underscores
  --> $DIR/literals.rs:54:18
   |
LL |     let fail22 = 3__4___23;
   |                  ^^^^^^^^^ help: consider: `3_423`

error: digits grouped inconsistently by underscores
  --> $DIR/literals.rs:55:18
   |
LL |     let fail23 = 3__16___23;
   |                  ^^^^^^^^^^ help: consider: `31_623`

error: mistyped literal suffix
  --> $DIR/literals.rs:57:18
   |
LL |     let fail24 = 12.34_64;
   |                  ^^^^^^^^ help: did you mean to write: `12.34_f64`

error: mistyped literal suffix
  --> $DIR/literals.rs:58:18
   |
LL |     let fail25 = 1E2_32;
   |                  ^^^^^^ help: did you mean to write: `1E2_f32`

error: mistyped literal suffix
  --> $DIR/literals.rs:59:18
   |
LL |     let fail26 = 43E7_64;
   |                  ^^^^^^^ help: did you mean to write: `43E7_f64`

error: mistyped literal suffix
  --> $DIR/literals.rs:60:18
   |
LL |     let fail27 = 243E17_32;
   |                  ^^^^^^^^^ help: did you mean to write: `243E17_f32`

error: mistyped literal suffix
  --> $DIR/literals.rs:61:18
   |
LL |     let fail28 = 241251235E723_64;
   |                  ^^^^^^^^^^^^^^^^ help: did you mean to write: `241_251_235E723_f64`

error: mistyped literal suffix
  --> $DIR/literals.rs:62:18
   |
LL |     let fail29 = 42279.911_32;
   |                  ^^^^^^^^^^^^ help: did you mean to write: `42_279.911_f32`

error: aborting due to 31 previous errors

